### PR TITLE
#599 Add action for mailbox

### PR DESCRIPTION
**Pull request by dab246**
https://user-images.githubusercontent.com/80730648/171166530-6e337ead-d04e-496d-9b46-22320d59ee51.mov

https://user-images.githubusercontent.com/80730648/171166932-8363c0b9-3f16-461b-9b0e-64291b2b3298.mov

### CODE REVIEW

**Comment by hoangdat** [comment]
> Given an unfolded mailbox tree, When I delete a mailbox, Then the tree automatically fold itself.
> 
> We likely should try to preserve tree folding IMO

yes, please keep the tree unfold like before we delete @dab246

**Comment by hoangdat** [comment]
https://user-images.githubusercontent.com/80730648/171166932-8363c0b9-3f16-461b-9b0e-64291b2b3298.mov

from time: 1 -> 5, I see you click `Inbox` but the search bar show the back button. What happen?

**Comment by dab246** [comment]
> 

It accidentally clicked the search button in the thread view before clicking on the mailbox name.

File: lib/features/mailbox/data/repository/mailbox_repository_impl.dart [hunk 1]
```diff
@@ -152,4 +171,59 @@ class MailboxRepositoryImpl extends MailboxRepository {
   Future<bool> renameMailbox(AccountId accountId, RenameMailboxRequest request) {
     return mapDataSource[DataSourceType.network]!.renameMailbox(accountId, request);
   }
+
+  @override
+  Future<bool> markAsMailboxRead(AccountId accountId, MailboxId mailboxId, MailboxName mailboxName) async {
```
Comment (hoangdat) [inline lib/features/mailbox/data/repository/mailbox_repository_impl.dart:176]: this is a heavy task in some case, how about this task block the main isolate for a while?

Comment (hoangdat) [reply lib/features/mailbox/data/repository/mailbox_repository_impl.dart:176]: I created a new ticket #616 to improve that.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**